pyiron/atomistics/job/interactive.py

```diff
@@ -270,65 +268,26 @@ def interactive_time_getter(self):
     def interactive_volume_getter(self):
         return self.initial_structure.get_volume()
 
-    def get_structure(self, iteration_step=-1, wrap_atoms=True):
+    def _update_previous_structure(self):
         """
-        Gets the structure from a given iteration step of the simulation (MD/ionic relaxation). For static calculations
-        there is only one ionic iteration step
+        Update the previous structure to the last step configuration
         Args:
-            iteration_step (int): Step for which the structure is requested
             wrap_atoms (bool):
-
-        Returns:
-            atomistics.structure.atoms.Atoms object
         """
-        if (
-            self.server.run_mode.interactive
-            or self.server.run_mode.interactive_non_modal
-        ):
-            # Warning: We only copy symbols, positions and cell information - no tags.
-            if self.output.indices is not None and len(self.output.indices) != 0:
-                indices = self.output.indices[iteration_step]
-            else:
-                return None
-            if len(self._interactive_species_lst) == 0:
-                el_lst = [el.Abbreviation for el in self.structure.species]
-            else:
-                el_lst = self._interactive_species_lst.tolist()
-            if indices is not None:
-                if wrap_atoms:
-                    positions = self.output.positions[iteration_step]
-                else:
-                    if len(self.output.unwrapped_positions) > max([iteration_step, 0]):
-                        positions = self.output.unwrapped_positions[iteration_step]
-                    else:
-                        positions = (
-                            self.output.positions[iteration_step]
-                            + self.output.total_displacements[iteration_step]
-                        )
-                atoms = Atoms(
-                    symbols=np.array([el_lst[el] for el in indices]),
-                    positions=positions,
-                    cell=self.output.cells[iteration_step],
-                    pbc=self.structure.pbc,
-                )
-                # Update indicies to match the indicies in the cache.
-                atoms.set_species([self._periodic_table.element(el) for el in el_lst])
-                atoms.indices = indices
-                if wrap_atoms:
-                    atoms = atoms.center_coordinates_in_unit_cell()
-                return atoms
-            else:
-                return None
+        try:
+            indices = self.output.indices[-1]
+            positions = self.output.positions[-1]
+            cell = self.output.cells[-1]
+        except IndexError:
+            return
+        if len(self._interactive_species_lst) == 0:
+            el_lst = [el.Abbreviation for el in self.structure.species]
```

Comment:
I thought that this was already available in `self.structure.get_chemical_symbols()`, but the implementation there is slightly different and uses the indices. I suspect that's redundant though, and if so you should just use the method instead of manually constructing the list.

Reply:
I don't know exactly how it's implemented, but pyiron has a quite complex convention when it comes to symbols/species/elements/indices. Some of them simply list elements, others per-atom elements, and sometimes they also account for original elements and not the user-defined elements etc. I took a brief look at different functions but it looks like the safest way is really to keep what was already in place.

Reply:
I 100% accept that rational for this PR, but because of this knowledge I just threw up a little in my mouth.